### PR TITLE
Revert "rbpf-cli: add coverage flag"

### PR DESCRIPTION
**Pull request by dmakarov**
This reverts commit 506ee0014a00f1943496592fb927075331ef66f3.

#### Problem

The original changes upgraded a ton of cargo crates. Update should be done more carefully.

#### Summary of Changes

Revert the 506ee0014a00f1943496592fb927075331ef66f3 and update cargo crates for the reverted Cargo.toml.

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
v1.10 too, please

**Comment by dmakarov** [comment]
> v1.10 too, please

#25411

**Review by CriesofCarrots — CHANGES_REQUESTED** [review]
This revert actually updates a bunch of dependencies again, instead of rolling them back

**Comment by dmakarov** [comment]
> This revert actually updates a bunch of dependencies again, instead of rolling them back

The commit, that I'm trying to revert, updated the following versions (should I reset them to the versions before the update? I'm not sure that many of these were not subsequently updated in more recent commits/PRs unrelated to the reverted commit and how this will affect the state of dependencies)

```
    Updating ansi_term v0.11.0 -> v0.12.1
    Updating async-stream v0.3.2 -> v0.3.3
    Updating async-stream-impl v0.3.2 -> v0.3.3
    Updating autocfg v1.0.1 -> v1.1.0
    Updating axum v0.5.1 -> v0.5.6
    Updating axum-core v0.2.1 -> v0.2.4
    Updating base64ct v1.3.3 -> v1.5.0
    Updating bindgen v0.59.1 -> v0.59.2
    Removing bitvec v0.19.5
    Updating block-buffer v0.10.0 -> v0.10.2
    Updating bumpalo v3.8.0 -> v3.9.1
    Updating camino v1.0.5 -> v1.0.9
    Updating cc v1.0.71 -> v1.0.73
    Updating cexpr v0.5.0 -> v0.6.0
    Updating clang-sys v1.2.2 -> v1.3.2
    Removing clap v2.33.3
    Removing clap v3.1.8
      Adding clap v2.34.0
      Adding clap v3.1.18
    Updating clap_derive v3.1.7 -> v3.1.18
      Adding clap_lex v0.2.0
    Updating cmake v0.1.46 -> v0.1.48
    Updating const_format v0.2.22 -> v0.2.23
    Updating core-foundation v0.9.2 -> v0.9.3
    Updating cpufeatures v0.2.1 -> v0.2.2
    Updating crc32fast v1.2.1 -> v1.3.2
    Updating crossbeam-epoch v0.9.5 -> v0.9.8
    Updating crossbeam-utils v0.8.5 -> v0.8.8
    Updating derive_more v0.99.16 -> v0.99.17
    Updating dialoguer v0.10.0 -> v0.10.1
    Updating ed25519 v1.2.0 -> v1.5.2
    Updating educe v0.4.18 -> v0.4.19
    Updating encoding_rs v0.8.29 -> v0.8.31
    Updating enum-ordinalize v3.1.10 -> v3.1.11
    Updating etcd-client v0.9.0 -> v0.9.2
      Adding fallible-iterator v0.2.0
    Updating fastrand v1.6.0 -> v1.7.0
    Updating filedescriptor v0.8.1 -> v0.8.2
    Updating filetime v0.2.15 -> v0.2.16
    Updating fixedbitset v0.4.0 -> v0.4.1
    Removing funty v1.1.0
    Updating getrandom v0.2.3 -> v0.2.6
      Adding gimli v0.26.1
    Updating h2 v0.3.11 -> v0.3.13
    Updating http v0.2.6 -> v0.2.7
    Updating httparse v1.5.1 -> v1.7.1
    Updating httpdate v1.0.1 -> v1.0.2
    Updating hyper v0.14.14 -> v0.14.18
    Updating im v15.0.0 -> v15.1.0
    Updating inout v0.1.2 -> v0.1.3
    Updating ipnet v2.3.1 -> v2.5.0
    Updating itoa v1.0.1 -> v1.0.2
    Updating libc v0.2.124 -> v0.2.126
    Updating libm v0.2.1 -> v0.2.2
    Updating libz-sys v1.1.3 -> v1.1.6
    Updating linux-raw-sys v0.0.42 -> v0.0.46
    Updating lock_api v0.4.6 -> v0.4.7
    Updating log v0.4.14 -> v0.4.17
    Updating memchr v2.4.1 -> v2.5.0
    Updating memoffset v0.6.4 -> v0.6.5
    Updating minimal-lexical v0.1.4 -> v0.2.1
    Updating mio v0.8.0 -> v0.8.3
    Updating native-tls v0.2.8 -> v0.2.10
    Updating nix v0.24.0 -> v0.24.1
    Removing nom v6.1.2
    Removing nom v7.0.0
      Adding nom v7.1.1
    Updating ntapi v0.3.6 -> v0.3.7
    Updating num-bigint v0.4.2 -> v0.4.3
    Updating num-integer v0.1.44 -> v0.1.45
    Updating num-traits v0.2.14 -> v0.2.15
    Updating num_threads v0.1.3 -> v0.1.6
    Updating once_cell v1.8.0 -> v1.11.0
    Updating openssl v0.10.38 -> v0.10.40
      Adding openssl-macros v0.1.0
    Updating openssl-probe v0.1.4 -> v0.1.5
    Updating openssl-src v300.0.5+3.0.2 -> v111.20.0+1.1.1o
    Updating openssl-sys v0.9.70 -> v0.9.73
    Updating os_str_bytes v6.0.0 -> v6.0.1
    Updating parking_lot_core v0.9.1 -> v0.9.3
    Updating pin-project v1.0.8 -> v1.0.10
    Updating pin-project-internal v1.0.8 -> v1.0.10
    Updating pin-project-lite v0.2.7 -> v0.2.9
    Updating pkg-config v0.3.22 -> v0.3.25
    Updating ppv-lite86 v0.2.15 -> v0.2.16
    Updating predicates v2.0.3 -> v2.1.1
    Updating predicates-core v1.0.2 -> v1.0.3
    Updating predicates-tree v1.0.4 -> v1.0.5
    Updating prettyplease v0.1.9 -> v0.1.10
    Updating proc-macro-crate v1.1.0 -> v1.1.3
    Updating proc-macro2 v1.0.32 -> v1.0.39
    Updating prost v0.10.0 -> v0.10.3
    Updating prost-build v0.10.0 -> v0.10.3
    Updating prost-derive v0.10.0 -> v0.10.1
    Updating prost-types v0.10.0 -> v0.10.1
    Updating quinn-udp v0.1.0 -> v0.1.1
    Updating quote v1.0.10 -> v1.0.18
    Removing radium v0.5.3
    Updating rand_xoshiro v0.4.0 -> v0.6.0
    Updating raptorq v1.6.5 -> v1.7.0
    Updating rayon v1.5.2 -> v1.5.3
    Updating rayon-core v1.9.2 -> v1.9.3
    Updating redox_syscall v0.2.10 -> v0.2.13
    Updating redox_users v0.4.0 -> v0.4.3
    Updating reed-solomon-erasure v5.0.1 -> v5.0.2
    Updating retain_mut v0.1.7 -> v0.1.9
    Removing rustc_version v0.3.3
    Updating rustix v0.34.3 -> v0.34.7
    Updating rustls v0.20.4 -> v0.20.6
    Updating rustls-native-certs v0.6.1 -> v0.6.2
      Adding rustls-pemfile v1.0.0
    Updating ryu v1.0.5 -> v1.0.10
    Updating schannel v0.1.19 -> v0.1.20
    Updating security-framework v2.4.2 -> v2.6.1
    Updating security-framework-sys v2.4.2 -> v2.6.1
    Removing semver v0.11.0
    Removing semver v1.0.7
      Adding semver v1.0.9
    Removing semver-parser v0.10.2
    Updating serde v1.0.136 -> v1.0.137
    Updating serde_bytes v0.11.5 -> v0.11.6
    Updating serde_derive v1.0.136 -> v1.0.137
    Updating serde_json v1.0.79 -> v1.0.81
    Updating serde_yaml v0.8.23 -> v0.8.24
    Updating signal-hook v0.3.13 -> v0.3.14
    Updating signature v1.4.0 -> v1.5.0
    Updating slab v0.4.5 -> v0.4.6
    Updating smallvec v1.7.0 -> v1.8.0
    Updating solana-frozen-abi v1.10.9 -> v1.10.17
    Updating solana-frozen-abi-macro v1.10.9 -> v1.10.17
    Updating solana-logger v1.10.9 -> v1.10.17
    Updating solana-program v1.10.9 -> v1.10.17
    Updating solana-sdk v1.10.9 -> v1.10.17
    Updating solana-sdk-macro v1.10.9 -> v1.10.17
    Updating spin v0.9.2 -> v0.9.3
    Updating syn v1.0.91 -> v1.0.95
    Removing tap v1.0.1
    Updating termcolor v1.1.2 -> v1.1.3
    Updating termtree v0.2.3 -> v0.2.4
    Updating thiserror v1.0.30 -> v1.0.31
    Updating thiserror-impl v1.0.30 -> v1.0.31
    Updating tikv-jemalloc-sys v0.4.2+5.2.1-patched.2 -> v0.4.3+5.2.1-patched.2
    Updating tikv-jemallocator v0.4.1 -> v0.4.3
    Updating tinyvec v1.5.0 -> v1.6.0
    Updating tokio v1.17.0 -> v1.18.2
    Updating tokio-io-timeout v1.1.1 -> v1.2.0
    Updating tokio-rustls v0.23.2 -> v0.23.4
    Removing tokio-util v0.6.9
    Removing tokio-util v0.7.1
      Adding tokio-util v0.6.10
      Adding tokio-util v0.7.2
    Updating toml v0.5.8 -> v0.5.9
    Updating tonic v0.7.1 -> v0.7.2
    Updating tonic-build v0.7.0 -> v0.7.2
    Updating tower-http v0.2.5 -> v0.3.3
    Updating tracing v0.1.29 -> v0.1.34
    Updating tracing-attributes v0.1.18 -> v0.1.21
    Updating tracing-core v0.1.21 -> v0.1.26
    Updating tracing-subscriber v0.3.7 -> v0.3.11
    Updating typenum v1.14.0 -> v1.15.0
    Updating unicode-bidi v0.3.7 -> v0.3.8
      Adding unicode-ident v1.0.0
    Updating unicode-xid v0.2.2 -> v0.2.3
    Updating utf8-width v0.1.5 -> v0.1.6
      Adding valuable v0.1.0
    Updating version_check v0.9.3 -> v0.9.4
      Adding wasi v0.11.0+wasi-snapshot-preview1
    Updating wasm-bindgen-futures v0.4.28 -> v0.4.30
    Updating web-sys v0.3.55 -> v0.3.57
    Updating webpki-roots v0.22.1 -> v0.22.3
    Updating which v4.2.2 -> v4.2.5
    Updating windows-sys v0.32.0 -> v0.36.1
    Updating windows_aarch64_msvc v0.32.0 -> v0.36.1
    Updating windows_i686_gnu v0.32.0 -> v0.36.1
    Updating windows_i686_msvc v0.32.0 -> v0.36.1
    Updating windows_x86_64_gnu v0.32.0 -> v0.36.1
    Updating windows_x86_64_msvc v0.32.0 -> v0.36.1
    Removing wyz v0.2.0
    Updating xattr v0.2.2 -> v0.2.3
    Updating zeroize_derive v1.2.0 -> v1.3.2
    Updating zstd v0.11.1+zstd.1.5.2 -> v0.11.2+zstd.1.5.2
    Updating zstd-safe v5.0.1+zstd.1.5.2 -> v5.0.2+zstd.1.5.2
```

**Comment by CriesofCarrots** [comment]
> should I reset them to the versions before the update?

My thinking is to leave the following as they are in master, and roll back everything else to the versions before the update:
1. Solana monorepo crates
2. Tokio crates
3. Anything that has been further bumped since #24655

We will probably get a bunch of dependabot PRs as a result, and can work through the dependencies one by one.

**Comment by dmakarov** [comment]
I updated the PR, resetting the previously updated versions to the precise versions that they were at before the commit that is being reverted. Some of the versions couldn't be reset back because of other dependencies, I guess.

**Comment by dmakarov** [comment]
should I merge?

**Comment by CriesofCarrots** [comment]
Looks like `syn`, `proc-macro2`, and `clap` are still being bumped by this PR. If you can undo those changes, okay with me with @jstarry 's r+

**Review by CriesofCarrots — DISMISSED** [review]

**Comment by jstarry** [comment]
Can you split this into two commits? One that only does the revert and keeps the conflicts.. and one that fixes the conflicts?

**Comment by dmakarov** [comment]
> Can you split this into two commits? One that only does the revert and keeps the conflicts.. and one that fixes the conflicts?

done

File: Cargo.lock [hunk 1]
```diff
@@ -82,9 +82,9 @@ checksum = "250f629c0161ad8107cf89319e990051fae62832fd343083bea452d93e2205fd"
 
 [[package]]
 name = "ansi_term"
-version = "0.11.0"
+version = "0.12.1"
```
Comment (jstarry) [inline Cargo.lock:85]: Please revert to 0.11.0

File: Cargo.lock [hunk 2]
```diff
@@ -349,9 +332,9 @@ dependencies = [
 
 [[package]]
 name = "bitvec"
-version = "0.19.5"
+version = "0.19.6"
```
Comment (jstarry) [inline Cargo.lock:335]: Please revert to 0.19.5

File: Cargo.lock [hunk 3]
```diff
@@ -397,9 +380,9 @@ dependencies = [
 
 [[package]]
 name = "block-buffer"
-version = "0.10.0"
+version = "0.10.2"
```
Comment (jstarry) [inline Cargo.lock:383]: Please revert to 0.10.0

File: Cargo.lock [hunk 4]
```diff
@@ -737,9 +697,9 @@ dependencies = [
 
 [[package]]
 name = "clap"
-version = "2.33.3"
+version = "2.34.0"
```
Comment (jstarry) [inline Cargo.lock:700]: Please revert to 2.33.3

File: Cargo.lock [hunk 5]
```diff
@@ -981,9 +930,9 @@ dependencies = [
 
 [[package]]
 name = "crossbeam-utils"
-version = "0.8.5"
+version = "0.8.8"
```
Comment (jstarry) [inline Cargo.lock:933]: Please revert to 0.8.5

File: Cargo.lock [hunk 6]
```diff
@@ -2530,9 +2435,9 @@ checksum = "2687e6cf9c00f48e9284cf9fd15f2ef341d03cc7743abf9df4c5f07fdee50b18"
 
 [[package]]
 name = "minimal-lexical"
-version = "0.1.4"
+version = "0.2.1"
```
Comment (jstarry) [inline Cargo.lock:2438]: Please revert to 0.1.4

File: Cargo.lock [hunk 7]
```diff
@@ -2669,13 +2552,12 @@ dependencies = [
 
 [[package]]
 name = "nom"
-version = "7.0.0"
+version = "7.1.1"
```
Comment (jstarry) [inline Cargo.lock:2555]: Please revert to 7.0.0

**Review by jstarry — COMMENTED** [review]
Too many version bumps to comment... can you please go through more carefully?

File: Cargo.lock (hunk 3 above)
Comment (dmakarov) [reply Cargo.lock]: This is required by digest v0.10.3 which is required by solana-sdk v1.11.0

File: Cargo.lock (hunk 5 above)
Comment (dmakarov) [reply Cargo.lock]: This is required by crossbeam-channel 0.5.4, required by solana-metrics 1.11.0

File: Cargo.lock [hunk 8]
```diff
@@ -3547,15 +3355,11 @@ dependencies = [
 
 [[package]]
 name = "quote"
-version = "1.0.10"
+version = "1.0.18"
```
Comment (dmakarov) [inline Cargo.lock:3358]: This is required by enum-iterator-derive v0.8.1 by enum-iterator v0.8.1 by solana-program-runtime v1.11.0

File: Cargo.lock (hunk 7 above)
Comment (dmakarov) [reply Cargo.lock]: done

File: Cargo.lock (hunk 6 above)
Comment (dmakarov) [reply Cargo.lock]: done

File: Cargo.lock (hunk 4 above)
Comment (dmakarov) [reply Cargo.lock]: done

File: Cargo.lock (hunk 2 above)
Comment (dmakarov) [reply Cargo.lock]: done

File: Cargo.lock (hunk 1 above)
Comment (dmakarov) [reply Cargo.lock]: done

**Comment by dmakarov** [comment]
I went through all newer versions and downgraded ones that didn't conflict with more recent updates on master, I believe.

**Comment by dmakarov** [comment]
finally passed all checks. ok to merge?

**Comment by jstarry** [comment]
@dmakarov I still see a number of tokio crates being upgraded, can you comment on whether those are necessary or can be downgraded as well?

**Comment by dmakarov** [comment]
> @dmakarov I still see a number of tokio crates being upgraded, can you comment on whether those are necessary or can be downgraded as well?

tokio was updated on master recently

File: Cargo.lock [hunk 9]
```diff
@@ -2420,15 +2331,9 @@ dependencies = [
 
 [[package]]
 name = "log"
-<<<<<<< HEAD
 version = "0.4.17"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "abb12e687cfb44aa40f41fc3978ef76448f9b6038cad6aef4259d3c095a2382e"
-=======
-version = "0.4.14"
```
Comment (jstarry) [inline Cargo.lock]: Why not downgrade to 0.4.14?

File: Cargo.lock [hunk 10]
```diff
@@ -2727,9 +2604,9 @@ dependencies = [
 
 [[package]]
 name = "num-traits"
-version = "0.2.15"
+version = "0.2.14"
```
Comment (jstarry) [inline Cargo.lock:2607]: Why did you downgrade to 0.2.14?

File: Cargo.lock [hunk 11]
```diff
@@ -4126,8 +3930,7 @@ dependencies = [
 
 [[package]]
 name = "semver"
-version = "1.0.7"
```
Comment (jstarry) [inline Cargo.lock]: Why not downgrade to 1.0.7?

File: Cargo.lock [hunk 12]
```diff
@@ -7080,12 +6801,9 @@ dependencies = [
 
 [[package]]
 name = "tokio-rustls"
-version = "0.23.3"
-=======
-version = "0.23.2"
->>>>>>> parent of 506ee0014a (rbpf-cli: add coverage flag)
+version = "0.23.4"
```
Comment (jstarry) [inline Cargo.lock:6804]: Why not rollback to 0.23.3?

File: Cargo.lock [hunk 13]
```diff
@@ -7153,9 +6871,9 @@ dependencies = [
 
 [[package]]
 name = "tokio-util"
-version = "0.7.1"
+version = "0.7.2"
```
Comment (jstarry) [inline Cargo.lock:6874]: Why upgrade to 0.7.2?

File: Cargo.lock [hunk 14]
```diff
@@ -7129,17 +6847,17 @@ dependencies = [
...
+ "tokio-rustls 0.23.4",
  "tungstenite",
  "webpki 0.22.0",
  "webpki-roots",
 ]
 
 [[package]]
 name = "tokio-util"
-version = "0.6.9"
+version = "0.6.10"
```
Comment (jstarry) [inline Cargo.lock:6858]: Why upgrade to 0.6.10?

File: Cargo.lock [hunk 15]
```diff
@@ -7031,9 +6759,9 @@ dependencies = [
 
 [[package]]
 name = "tokio-io-timeout"
-version = "1.1.1"
+version = "1.2.0"
```
Comment (jstarry) [inline Cargo.lock:6762]: Why upgrade to 1.2.0?

File: Cargo.lock (hunk 8 above)
Comment (jstarry) [reply Cargo.lock:3358]: Sounds good, this impacts proc-macro2 as well

**Comment by jstarry** [comment]
> > @dmakarov I still see a number of tokio crates being upgraded, can you comment on whether those are necessary or can be downgraded as well?
> 
> tokio was updated on master recently

I couldn't find that commit, can you point me to it?

File: Cargo.lock (hunk 9 above)
Comment (dmakarov) [reply Cargo.lock]:
```
$ cargo update -p log --precise 0.4.14
    Updating crates.io index
error: failed to select a version for the requirement `log = "^0.4.17"`
candidate versions found which didn't match: 0.4.14
location searched: crates.io index
required by package `solana-frozen-abi v1.11.0 (/Users/dmakarov/work/git/solana/frozen-abi)`
```

File: Cargo.lock (hunk 11 above)
Comment (dmakarov) [reply Cargo.lock]:
```
$ cargo update -p semver:1.0.9 --precise 1.0.7
    Updating crates.io index
error: failed to select a version for the requirement `semver = "^1.0.9"`
candidate versions found which didn't match: 1.0.7
location searched: crates.io index
required by package `solana-version v1.11.0 (/Users/dmakarov/work/git/solana/version)`
```

**Review by jstarry — APPROVED** [review]

File: Cargo.lock (hunk 10 above)
Comment (dmakarov) [reply Cargo.lock]: > > > @dmakarov I still see a number of tokio crates being upgraded, can you comment on whether those are necessary or can be downgraded as well?
> > 
> > 
> > tokio was updated on master recently
> 
> I couldn't find that commit, can you point me to it?

my mistake. I assumed it was based on previous comments in this PR.

**Pull request merged**